includes/class-sensei-course.php

```diff
@@ -3459,6 +3475,103 @@ public function log_initial_publish_event( $course ) {
 		sensei_log_event( 'course_publish', $event_properties );
 	}
 
+	/**
+	 * Mark saving course id when no resave is needed for id sync.
+	 *
+	 * hooked into `save_post_course`.
+	 *
+	 * @since 3.6.0
+	 * @access private
+	 *
+	 * @param int      $post_id Post ID.
+	 * @param \WP_Post $post    Post object.
+	 */
+	public function mark_saving_course_id( $post_id, $post ) {
+		$content = $post->post_content;
+
+		if ( has_block( 'sensei-lms/course-outline', $content ) ) {
+			$blocks = parse_blocks( $content );
+
+			if ( ! $this->has_pending_id_sync( $blocks ) ) {
+				$this->course_id_saving = $post_id;
+			}
+		} else {
+			$this->course_id_saving = $post_id;
+		}
+	}
+
+	/**
+	 * Check if blocks has pending id sync.
+	 *
+	 * @since 3.6.0
+	 *
+	 * @param array[] $blocks Blocks array.
+	 *
+	 * @return boolean Whether has block with pending id sync.
+	 */
+	private function has_pending_id_sync( $blocks ) {
+		foreach ( $blocks as $block ) {
+			$is_checkable_block = 'sensei-lms/course-outline-module' === $block['blockName']
+				|| 'sensei-lms/course-outline-lesson' === $block['blockName'];
+
+			if (
+				// Check pending id sync.
+				( $is_checkable_block && empty( $block['attrs']['id'] ) )
+				// Check inner blocks pending id sync.
+				|| (
+					! empty( $block['innerBlocks'] )
+					&& $this->has_pending_id_sync( $block['innerBlocks'] )
+				)
+			) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Log the course update.
+	 *
+	 * hooked into `shutdown`
+	 *
+	 * @since 3.6.0
+	 * @access private
+	 */
+	public function log_course_update() {
+		if ( empty( $this->course_id_saving ) ) {
+			return;
+		}
+
+		$course_id     = $this->course_id_saving;
+		$post          = get_post( $course_id );
+		$content       = $post->post_content;
+		$product_ids   = get_post_meta( $course_id, '_course_woocommerce_product', false );
+		$product_count = empty( $product_ids ) ? 0 : count( array_filter( $product_ids, 'is_numeric' ) );
+
+		$event_properties = [
+			'course_id'         => $course_id,
+			'has_outline_block' => has_block( 'sensei-lms/course-outline', $content ) ? 1 : 0,
+			'module_count'      => count( wp_get_post_terms( $course_id, 'module' ) ),
+			'lesson_count'      => $this->course_lesson_count( $course_id ),
+			'product_count'     => $product_count,
```

Comment:
We could create a filter for that, and add this feature through the WCPC, but as it only for log I think it wouldn't make sense. And it follows the same idea used in the `sensei_course_publish `